src/flint/types/fq_default.pyx

```diff
@@ -0,0 +1,790 @@
+
+        """
+        cdef fmpz_mod_poly_ctx ctx
+        cdef fmpz_mod_poly pol
+        ctx = fmpz_mod_poly_ctx(self.prime())
+        pol = ctx.new_ctype_poly()
+        fq_default_ctx_modulus(pol.val, self.val)
+        return pol
+
+    def zero(self):
+        """
+        Return the zero element
+
+            >>> gf = fq_default_ctx(5)
+            >>> gf.zero()
+            0
+        """
+        cdef fq_default res
+        res = self.new_ctype_fq_default()
+        res.ctx = self
+        fq_default_zero(res.val, self.val)
+        return res
+
+    def one(self):
+        """
+        Return the unit element
+
+            >>> gf = fq_default_ctx(5)
+            >>> gf.one()
+            1
+        """
+        cdef fq_default res
+        res = self.new_ctype_fq_default()
+        res.ctx = self
+        fq_default_one(res.val, self.val)
+        return res
+
+    def gen(self):
+        """
+        Return the one element
+
+            >>> gf = fq_default_ctx(5, 2, var="w")
+            >>> gf.gen()
+            w
+        """
+        cdef fq_default res
+        res = self.new_ctype_fq_default()
+        res.ctx = self
+        fq_default_gen(res.val, self.val)
+        return res
+
+    def random_element(self, not_zero=False):
+        r"""
+        Return a random element of the finite field
+
+            >>> gf = fq_default_ctx(163, 3)
+            >>> a = gf.random_element()
+            >>> type(a) is fq_default
+            True
+            >>> a = gf.random_element(not_zero=True)
+            >>> not a.is_zero()
+            True
+        """
+        cdef fq_default res
+        res = self.new_ctype_fq_default()
+        res.ctx = self
+        if not_zero:
+            fq_default_rand_not_zero(res.val, global_random_state, self.val)
+        else:
+            fq_default_rand(res.val, global_random_state, self.val)
+        return res
+
+    cdef new_ctype_fq_default(self):
+        return fq_default.__new__(fq_default, None, self)
+
+    cdef set_list_as_fq_default(self, fq_default_t fq_ele, obj):
+        cdef fmpz_poly poly
+        poly = fmpz_poly.__new__(fmpz_poly)
+        fmpz_poly_set_list(poly.val, obj)
+
+        # Now set the value from the fmpz_poly
+        fq_default_set_fmpz_poly(fq_ele, poly.val, self.val)
+
+        return 0
+
+    cdef set_any_as_fq_default(self, fq_default_t fq_ele, obj):
+        # Converts the list to an fmpz_poly and then sets from this
+        if typecheck(obj, list):
+            return self.set_list_as_fq_default(fq_ele, obj)
+
+        # For small integers we can convert directly
+        if typecheck(obj, int) and obj.bit_length() < 32:
+            fq_default_set_si(fq_ele, <slong>obj, self.val)
+            return 0
+
+        # Assumes that the modulus of the polynomial matches
+        # the context for the fq_default
+        if typecheck(obj, fmpz_mod_poly):
+            fq_default_set_fmpz_mod_poly(fq_ele, (<fmpz_mod_poly>obj).val, self.val)
+            return 0
```

Comment:
If the modulus doesn't match then what happens? Does it just set without checking or does it crash or something?

Reply:
I think it simply takes the polynomial and ignores the modulus reducing modulo the fq context instead